diff --git a/pyslam/slam/cpp/utils/descriptor_helpers.h b/pyslam/slam/cpp/utils/descriptor_helpers.h
index efc60ae..379576e 100644
--- a/pyslam/slam/cpp/utils/descriptor_helpers.h
+++ b/pyslam/slam/cpp/utils/descriptor_helpers.h
@@ -253,7 +253,8 @@ static inline float hsum512_ps(__m512 v) noexcept {
     s = _mm_hadd_ps(s, s);
     return _mm_cvtss_f32(s);
 }
-#elif defined(__AVX2__)
+#endif
+#if defined(__AVX2__)
 #include <immintrin.h>
 static inline float hsum256_ps(__m256 v) noexcept {
     __m128 low = _mm256_castps256_ps128(v);
@@ -265,7 +266,8 @@ static inline float hsum256_ps(__m256 v) noexcept {
     sums = _mm_add_ss(sums, shuf);       // s3+s2+s1+s0
     return _mm_cvtss_f32(sums);
 }
-#elif defined(__ARM_NEON)
+#endif
+#if defined(__ARM_NEON)
 #include <arm_neon.h>
 #endif
 
